### DISCUSSION TITLE
Slow Unit Tests

### DISCUSSION THREAD

**Discussion by jeswheel** [pypomp/pypomp#3] [General]
According to [unit-test statistics](https://app.codecov.io/gh/pypomp/pypomp/tests/main) provided by Codecov, some of our unit-tests take longer than 10 seconds to run on average. This seems a bit slow, should we think about how to speed these up in the future?

**Comment by ionides** [comment]
I agree. Perhaps reducing the numbers of particles, iterations and time series length will be sufficient. We can look for a volunteer to do this in our meeting tomorrow.

**Reply by junch2002** [reply]
I previously set the number of particles J = 10, iterations M = 2, and time series length T = 4. It is a simple model, but I can further set the number of particles J = 3. Based on Jesse's suggestion, I'm trying to switch to Pytest using fixtures from unittest and see how it performs. I can also find to delete some redundant tests as well.

**Comment by jeswheel** [comment]
One way to improve tests in general (for both speed and syntax) is the use of [Pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html). I'm already using Pytest in the GitHub actions, so adding them directly to the tests shouldn't be too much of a change. @junch2002, what do you think?

**Reply by junch2002** [reply]
That sounds to be a great way. I'll try that!
